web/controllers/screeningController/index.ts

```diff
@@ -347,6 +374,67 @@ export async function updateCourse(req: Request, res: Response) {
     }
 }
 
+async function handleNewLectures(lectures: { subcourse: { id: number }, start: Date, duration: number, instructor: { id: number } }[], courseId: number) {
+    const entityManager = getManager();
+
+    for (let i=0; i<lectures.length; i++){
+        const course = await entityManager.findOne(Course, { id: courseId });
+        if (course == undefined) {
+            logger.warn(`No course found with ID ${courseId}`);
+            return 404;
+        }
+        const subcourse = await entityManager.findOne(Subcourse, { id: lectures[i].subcourse.id, course: course });
+        if (subcourse == undefined) {
+            logger.warn(`No subcourse found with ID ${lectures[i].subcourse.id}`);
+            return 404;
+        }
+        const instructor = await entityManager.findOne(Student, { id: lectures[i].instructor.id });
```

Comment:
same here, `await subcourse.getInstructor()` could be better, especially if the db model in the future.

Reply:
There is exactly one instructor specified for each lecture while there can be many for a subcourse. Hence extracting the instructor from the subcourse wouldn't make sense here. Nevertheless, in the lecture constructor I added to the subcourse I take the instructor from the subcourse matching the given instructor id.

Reply:
Yeah, your new version looks reasonable.